### ISSUE TITLE
Posts didn't show up after changing PDS domain

### ISSUE THREAD

**Issue by astrojuanlu** [bluesky-social/social-app#5446]
**Describe the bug**

I suspect there's a bug in the Bluesky app when a custom PDS changes its domain name.

I was running a custom PDS on `bsky.juanlu.space`, everything was working well.

At some point I decided to change the domain name, so I added a DSN A record `pds.juanlu.space` pointing to the same IP, changed the `Caddyfile` and `pds.env`, restarted the service. I was able to log in and browse the network, so for a few days I thought everything had worked fine.

However, at some point I found out that my posts weren't being shown. I would refresh my profile page and see nothing.

I started digging. The `POST /xrpc/com.atproto.repo.createRecord` was returning a valid record with `"validationStatus":"valid"`. The `/xrpc/com.atproto.sync.subscribeRepos` WebSocket was showing my new posts. And yet, new posts weren't showing in the app.

I also noticed that the app wasn't able to retrieve my avatar. But the avatar for old posts was showing fine.

I am not a TypeScript coder nor ATProto expert so it would have taken me a while to debug this without assistance. So I decided to revert to the last working configuration hoping that I hadn't broken anything.

I changed the DNS back to bsky.juanlu.space, reverted the configs, restarted the service, and the app caught up with all the posts that I had written. So now everything is back to normal.

**To Reproduce**

(see above)

**Expected behavior**

Everything should continue as normal even after changing the domain name.

But maybe this is either (b) an improper expectation because $cryptography or whatever or (c) I did miss some config that I should have changed, so error was between chair and keyboard. Happy to dig a bit more with some assistance (I'm fine using my own account as guinea pig).

**Screenshots**

**Details**

 - Platform: Firefox on desktop Linux Mint
 - Platform version: 130.0
 - App version: Build version: 1.91.1; Bundle info: 49ab486 (prod); Bundle date: 24092017; Platform: web

**Additional context**

**Comment by azurebluemax** [comment]
> this migration guide

To clarify:

The "old pds" steps would need to be done with the config on the previous domain, and the "new pds" step is after changing the domain?? (Since technically, it's all the same PDS so there isn't an old and new in this use case...)

**Comment by mary-ext** [comment]
yeah old and new PDS refers to the same PDS in this case, since we just want to update the location of the PDS

**Comment by hackdefendr** [comment]
All set, I did only the steps from your screenshot and it fixed my posting issue. Of course I had to login first 🥇 .

**Comment by mary-ext** [comment]
For anyone tuning in to this issue, there's now an easier way of doing it via https://boat.kelinci.net/plc-applicator

When you login, you want to make sure that it's pointing at the new PDS hostname, and then you can either edit it yourself or click on "Use PDS recommendation" to get the correct one.

**Comment by astrojuanlu** [comment]
This worked perfectly, first time ✔️ Thanks @mary-ext! The devs can keep this issue open or close it as they wish

**Comment by ontheair81** [comment]
This helped me a lot. Thank you very much!
